File: core/src/main/java/hello/core/discount/implement/FixDiscountPolicy.java
package hello.core.discount.implement;

import hello.core.discount.DiscountPolicy;
import hello.core.member.data.Grade;
import hello.core.member.data.Member;

public class FixDiscountPolicy implements DiscountPolicy {

    private int discountFixAmount = 1000;

    @Override
    public int discount(Member member, int price) {
        // enum 타입은 == 쓰는게 맞음
        if (member.getGrade() == Grade.VIP) {
            return discountFixAmount;
        } else {
            return 0;
        }
    }
}

File: core-kotlin/src/main/kotlin/hello/corekotlin/discount/implement/FixDiscountPolicy.kt
package hello.corekotlin.discount.implement

import hello.corekotlin.discount.DiscountPolicy
import hello.corekotlin.member.data.Grade
import hello.corekotlin.member.data.Member

class FixDiscountPolicy: DiscountPolicy {

    private val discountFixAmount = 1000

    override fun discount(member: Member, price: Int): Int {
        return member.run {
            if (this.grade == Grade.VIP) {
                discountFixAmount
            } else {
                0
            }
        }
    }
}